nav2_behavior_tree/test/plugins/action/test_compute_path_to_pose_action.cpp

```diff
@@ -41,8 +41,13 @@ class ComputePathToPoseActionServer : public TestActionServer<nav2_msgs::action:
   {
     const auto goal = goal_handle->get_goal();
     auto result = std::make_shared<nav2_msgs::action::ComputePathToPose::Result>();
-    result->path.poses.resize(1);
-    result->path.poses[0].pose.position.x = goal->pose.pose.position.x;
+    result->path.poses.resize(2);
```

Comment:
This is not a test that this works in the planner, just that the behavior tree responds properly. You must test using the actual planner server, please update the system tests for planning. This can also be helpful, but this isn't a test that these changes are actually functional since this just mocks up the planner server interface

Reply:
> You can do this using the `system_tests/planner` tests. There's a random and fixed map tests you could add an additional test case to where you set `use_start` and a `start` to show that it respects that and not the current robot TF starting pose.

It is more complex than that, as I tried to express above, the current `system_tests/planner` tests are never testing the `ComputePathToPose` `action_server` of the `planner_server` defined here:
https://github.com/ros-planning/navigation2/blob/2d88d14b1dd9f678e9b95dd50c5db774667cd62e/nav2_planner/src/planner_server.cpp#L132-L136

And hence, the logic of getting the pose of the robot as a start of the path (or the `use_start` potential override) as implemented in the action server of planner_server is never tested.
Instead, it is the only the `createPlan(start, goal)` fonction of the `planner_server` which is tested, not the full action. And the `planner_tester` implements its own logic for retrieving the start pose of the path, see here:
https://github.com/ros-planning/navigation2/blob/2d88d14b1dd9f678e9b95dd50c5db774667cd62e/nav2_system_tests/src/planning/planner_tester.hpp#L77-L86

What we need, I believe, is to create tests calling the `ComputePathToPose` `action_server` of the `planner_server`, for both `use_start` `true` and `false`. I might consider doing it if align and if you give me pointers on the architecture of it. Should `planner_tester` be extended to call the action ? Or a separate class ?

Reply:
Got it, thanks for the explanation. I agree those tests are not probably in the structure for this testing.

The system tests [have this structure](https://github.com/ros-planning/navigation2/blob/main/nav2_system_tests/src/system/test_system_launch.py#L108-L118) of launching a launch description (which could just contain a lifecycle manager to transition up the planner server and the planner server) and then a test python node to trigger inputs and outputs. That could work here but the issue I see is that no info would actually come to the planner server, so you'd need to also have a `map_server` to publish some map for the planner server's costmap to be aware of. Then when you called `ComputePathToPose` you should be good to go (but probably have to fake out TF).

It turns into a bit of a challenge. I could wave the testing of this one in CI if you verified locally that this worked fully. Though it might be a good exercise if you haven't done something like that before and I would never argue with more :smile:

Reply:
Hum, it will be indeed a good exercise. No promise as I am very busy atm, but don't hesitate to ping me in a week or so if you see no progress (and yes, it was tested locally).